From 96b67bab47ba1ab27e121b07b7cbdcabe7b2e6f2 Mon Sep 17 00:00:00 2001
From: Pawel MOLL <pawel.moll@st.com>
Date: Tue, 3 Mar 2009 14:38:48 +0000
Subject: [PATCH] sh_stm: Fix ethernet and audio configuration in PDK7105 setup.

This patch fixes the ethernet configuration on PDK7105 board and
configures audio pins as well (I2S output and inputs are available
on JS1 header).

Signed-off-by: Pawel Moll <pawel.moll@st.com>
---
 arch/sh/boards/st/pdk7105/setup.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/sh/boards/st/pdk7105/setup.c b/arch/sh/boards/st/pdk7105/setup.c
index bdf2907..c6a9476 100644
--- a/arch/sh/boards/st/pdk7105/setup.c
+++ b/arch/sh/boards/st/pdk7105/setup.c
@@ -102,7 +102,7 @@ static int pdk7105_phy_reset(void* bus)
 static struct plat_stmmacphy_data phy_private_data = {
 	/* Micrel */
 	.bus_id = 0,
-	.phy_addr = 0,
+	.phy_addr = -1,
 	.phy_mask = 0,
 	.interface = PHY_INTERFACE_MODE_MII,
 	.phy_reset = &pdk7105_phy_reset,
@@ -161,8 +161,9 @@ static int __init device_init(void)
 
 	phy_reset_pin = stpio_request_set_pin(15, 5, "eth_phy_reset",
 					      STPIO_OUT, 1);
-	stx7105_configure_ethernet(0, 0, 0, 1, 0, 0);
+	stx7105_configure_ethernet(0, 0, 0, 0, 0, 0);
 	stx7105_configure_lirc(&lirc_scd);
+	stx7105_configure_audio_pins(3, 1, 1);
 
 	return platform_add_devices(pdk7105_devices, ARRAY_SIZE(pdk7105_devices));
 }
-- 
1.6.0.6

